src/index.tsx

```diff
@@ -0,0 +1,140 @@
+import invariant from 'invariant';
+import React, { useCallback, useContext, useState } from 'react';
+import { ReactRelayContext, commitMutation } from 'react-relay';
+import {
+  MutationConfig as BaseMutationConfig,
+  Environment,
+  OperationBase,
+} from 'relay-runtime';
+
+import { Omit, WithOptionalFields } from './typeHelpers';
+
+export type MutationState<T extends OperationBase> = {
+  loading: boolean;
+  data: T['response'] | null;
```

Comment:
should this just be `T`? or am i misunderstanding the type def

Reply:
T has two fields, `variables` and `response`. we want response here